zerver/lib/user_topics.py

```diff
@@ -96,39 +97,45 @@ def set_topic_mutes(
         recipient_id = stream.recipient_id
         assert recipient_id is not None
 
-        add_topic_mute(
+        add_topic_visibility_policy(
             user_profile=user_profile,
             stream_id=stream.id,
             recipient_id=recipient_id,
             topic_name=topic_name,
-            date_muted=date_muted,
+            visibility_policy=UserTopic.MUTED,
+            date_added=date_muted,
         )
 
 
-def add_topic_mute(
+def add_topic_visibility_policy(
     user_profile: UserProfile,
     stream_id: int,
     recipient_id: int,
     topic_name: str,
-    date_muted: Optional[datetime.datetime] = None,
+    visibility_policy: int,
+    date_added: Optional[datetime.datetime] = None,
     ignore_duplicate: bool = False,
 ) -> None:
-    if date_muted is None:
-        date_muted = timezone_now()
-    UserTopic.objects.bulk_create(
-        [
-            UserTopic(
-                user_profile=user_profile,
-                stream_id=stream_id,
-                recipient_id=recipient_id,
-                topic_name=topic_name,
-                last_updated=date_muted,
-                visibility_policy=UserTopic.MUTED,
-            ),
-        ],
-        ignore_conflicts=ignore_duplicate,
+    if date_added is None:
+        date_added = timezone_now()
+
+    (row, created) = UserTopic.objects.get_or_create(
+        user_profile=user_profile,
+        stream_id=stream_id,
+        recipient_id=recipient_id,
+        topic_name=topic_name,
+        defaults={"last_updated": date_added, "visibility_policy": visibility_policy},
     )
```

Comment:
What's the motivation for the change away from `bulk_create`? I worry about this accidentally reverting the work in 781107308da36f3a5ab70408a0f95acb3b2a82e4.

Reply:
I have reverted it back to `bulk_create`. [Django Docs](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#bulk-create) for the new parameters used.